spec/loophp/collection/CollectionSpec.php

```diff
@@ -60,6 +61,22 @@ public function it_can_all(): void
             ->shouldIterateAs(['a' => 3, 'b' => 2]);
     }
 
+    public function it_fails_to_work_with_generator(): void
+    {
+        $gen = static fn (): Generator => yield from [1, 2, 3];
+
+        $this->beConstructedThrough('fromIterable', [$gen()]);
+        $this->shouldHaveCount(3);
+        $this->shouldIterateAs([1, 2, 3]);
+    }
```

Comment:
looks like the issue is here, when we instantiate the collection from a generator, which is what also happens in `partition`